src/libraries/Microsoft.PowerFx.Core/Public/Config/Features.cs

```diff
@@ -82,6 +82,11 @@ public sealed class Features
         /// </summary>
         internal bool IsUserDefinedTypesEnabled { get; init; } = false;
 
+        /// <summary>
+        /// Enables treating void type expression as error in non-side-effecting expressions <see cref="ParserOptions.AllowsSideEffects"/>.
+        /// </summary>
+        internal bool IsVoidError { get; init; } = false;
```

Comment:
Do we need this at all? Given that any void expressions in a context where bindingContext.AllowsSideEffects is false today may be a runtime error (it would now become a compile-time error). While this is technically a breaking change, it looks like one that we should make.

CC: @MikeStall, @gregli-msft